src/components/modal/ModalComposite.tsx

```diff
@@ -23,6 +24,7 @@ export interface IModalCompositeOwnProps extends IModalOwnProps, IModalHeaderOwn
     modalFooterChildren?: React.ReactNode;
     modalFooterClasses?: IClassName;
     isPrompt?: boolean;
+    validateShouldNavigate?: (isDirty: boolean) => void;

```

Comment:
`validateShouldNavigate` must return a boolean
```suggestion
    validateShouldNavigate?: (isDirty: boolean) => boolean;
```